sdk/resource/env.go

```diff
@@ -88,7 +89,12 @@ func constructOTResources(s string) (*Resource, error) {
 			invalid = append(invalid, p)
 			continue
 		}
-		k, v := strings.TrimSpace(field[0]), strings.TrimSpace(field[1])
+		k := strings.TrimSpace(field[0])
+		v, err := url.QueryUnescape(strings.TrimSpace(field[1]))
+		if err != nil {
+			// Retain original value if decoding fails.
+			v = field[1]
+		}
```

Comment:
The spec actually defines that the invalid character should be replaced 

> When decoding the value, percent-encoded octet sequences that do not match the UTF-8 encoding scheme MUST be replaced with the replacement character (U+FFFD).

This would take quite a bit of extra work, so I thought it would be better to do it in the `baggage` package and then migrate the SDK to use it?